src/target/riscv/riscv.c

```diff
@@ -3361,7 +3373,8 @@ int riscv_openocd_step(struct target *target, int current,
 		LOG_TARGET_ERROR(target, "Unable to restore the disabled breakpoint.");
 	}
 
-	if (success) {
+	// TODO: should we update debug_reason if no callback was called?
```

Comment:
I think that we should update the debug reason even if callbacks are not to be handled.

Reply:
addressed

Reply:
@MarekVCodasip Marek, can you please quickly re-check this PR after your suggestion has been addressed. Thanks.